kubeconfig_plugin.py

```diff
@@ -117,57 +138,106 @@ class ErrorOutput:
     outputs={"success": SuccessOutput, "error": ErrorOutput},
 )
 def extract_kubeconfig(
-    params: InputParams,
+        params: InputParams,
 ) -> typing.Tuple[str, typing.Union[SuccessOutput, ErrorOutput]]:
-
     print("==>> Parsing and extracting kubernetes cluster details ...")
 
     try:
         kubeconfig = yaml.safe_load(params.kubeconfig)
-        kcl = config.kube_config.KubeConfigLoader(kubeconfig)
 
-        output = {}
-        output["cluster_name"] = kcl._current_context["context"]["cluster"]
-        output["server_url"] = kcl._cluster.value["server"]
-        output["certificate_authority_data"] = kcl._cluster.value[
-            "certificate-authority-data"
-        ]
-        output["user"] = kcl._current_context["context"]["user"]
+        try:
+            if kubeconfig["kind"] != "Config":
+                return "error", ErrorOutput("The provided file is not a kubeconfig file")
+        except KeyError:
+            return "error", ErrorOutput("The provided file is not a kubeconfig file (missing 'kind' field)")
 
         try:
-            output["client_certificate_data"] = kcl._user.value[
-                "client-certificate-data"
-            ]
-            output["client_key_data"] = kcl._user.value["client-key-data"]
-        except Exception:
-            print(
-                "client certificate and key data missing, trying to extract"
-                " token"
-            )
+            current_context = kubeconfig["current-context"]
+        except KeyError:
+            return "error", ErrorOutput(
+                "The provided kubeconfig file does not have a current-context set. Please set a current context to use.")
 
-        if (
-            "client_key_data" not in output
-            and "client_certificate_data" not in output
-        ):
-            try:
-                output["token"] = kcl._user.value["token"]
-            except Exception:
-                print("token missing for user in kubeconfig")
-
-        if (
-            "client_key_data" not in output
-            and "client_certificate_data" not in output
-            and "token" not in output
-        ):
+        try:
+            context = None
+            for ctx in kubeconfig["contexts"]:
+                if ctx["name"] == current_context:
```

Comment:
Just to make sure: `current_context` from `kubeconfig["current-context"] is a `string` and not a `dict`, right?